.github/workflows/pr.yml

```diff
@@ -36,6 +36,17 @@ jobs:
       python_version: ${{ needs.pyversion.outputs.pyversion }}
       poetry_version: 1.4.2
 
+  unit_tests:
+    name: Unit Tests
+    needs: [pyversion]
+    uses: ./.github/workflows/sub_unittest.yml
+    with:
+      python_version: ${{ needs.pyversion.outputs.pyversion }}
+      poetry_version: 1.4.2
+    permissions:
```

Comment:
as discussed on Monday, although this is sufficient I still think specifying these permissions in the sub-workflow is somehow self-documenting.

Reply:
ok, not pushing just an idea, not sure it will work, but how about this:

```yaml
permissions:
  contents: read
  pull-requests: ${{ github.event_name == 'pull_request'  && 'none' || 'write' }}
```

or something similar.

This is of course in the sub-workflow

Reply:
I tried with and without double quotes for conditionally setting permissions but it didn't work producing the below output:

```
The workflow is not valid. In .github/workflows/pr.yml (Line: 44, Col: 11): Error from called workflow PaloAltoNetworks/pan-os-upgrade-assurance/.github/workflows/sub_unittest.yml@f95643c8c1e137b67990eeb89983fd979e13dfa6 (Line: 22, Col: 22): Unrecognized named-value: 'github'. Located at position 1 within expression: github.event_name == 'pull_request' && 'write' || 'none' PaloAltoNetworks/pan-os-upgrade-assurance/.github/workflows/sub_unittest.yml@f95643c8c1e137b67990eeb89983fd979e13dfa6 (Line: 22, Col: 22): Unexpected value '${{ github.event_name == 'pull_request' && 'write' || 'none' }}'
```

Reply:
Looks like setting permissions conditionally is not supported. No permissions are specified in the reusable workflow.